src/definitions.jl

```diff
@@ -271,7 +271,15 @@ summary(p::ScaledPlan) = string(p.scale, " * ", summary(p.p))
 *(p::Plan, I::UniformScaling) = ScaledPlan(p, I.λ)
 
 # Normalization for ifft, given unscaled bfft, is 1/prod(dimensions)
-normalization(::Type{T}, sz, region) where T = one(T) / Int(prod(sz[r] for r in region))::Int
+# ensure that region is a subset of eachindex(sz).
+_checkindex(szinds, region::AbstractVector) = checkindex(Bool, szinds, region)
+# this method handles the case where region is not an array, e.g. it is a Tuple
+_checkindex(szinds, region) = all(r -> checkindex(Bool, szinds, r), region)
+@inline function normalization(::Type{T}, sz, region) where T
+    @boundscheck (!isempty(region) && _checkindex(eachindex(sz), region)) ||
+        throw(BoundsError(sz, region))
+    one(T) / mapreduce(r -> Int(@inbounds sz[r])::Int, *, region; init=oneunit(eltype(sz)))::Int
```

Comment:
If we are casting things to `Int`, shouldn't we use `init=1`?

Reply:
Updated